chainstate/src/detail/transaction_verifier/mod.rs

```diff
@@ -572,13 +520,14 @@ impl<'a, S: BlockchainStorageRead> TransactionVerifier<'a, S> {
             .undo
     }
 
-    pub fn connect_transaction(
+    pub fn connect_transactable(
         &mut self,
         spend_ref: BlockTransactableRef,
         spend_height: &BlockHeight,
         median_time_past: &BlockTimestamp,
         blockreward_maturity: &BlockDistance,
-    ) -> Result<(), ConnectTransactionError> {
+    ) -> Result<Option<Fee>, ConnectTransactionError> {
+        let mut fee: Option<Fee> = None;
```

Comment:
Nope... no mutable variables.

Reply:
I pushed a fix.

Please always avoid mutable variables. They're usually the source of undefined behavior. We only use mutable under heavily controlled environments and only when we must.